### PR TITLE
Added regression functional tests (closes #295)

### LINKED ISSUE #295: Add regression functional tests

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @AlexanderMoskovkin, @helen-dikareva

### CODE REVIEW

**Comment by helen-dikareva** [comment]
@VasilyStrelyaev please check tests

**Comment by helen-dikareva** [comment]
I don't like naming like this 'T171129-index.html'

**Comment by AlexanderMoskovkin** [comment]
Now in the `test/functional/fixtures/regression` there are:
`gh-414`
`gh-418`
`pages`
`testcafe-fixtures`
`test.js`

It's not good because we mix different type of folders - separate issues (the first two) and files/folders for other tests.
I suppose we should separate new regression tests with folders. And we will have
```
gh-414
gh-418
t116171
t171129
...
focus-blur-change
prevent-real-action
...

```

File: test/functional/fixtures/regression/testcafe-fixtures/prevent-real-action-in-same-domain.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'@fixture prevent real action in same domain page';
+'@page ./prevent-real-action-same-domain.html';
+
+
+'@test'['Type and wait in iframe with obsolete manner'] = {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/regression/testcafe-fixtures/prevent-real-action-in-same-domain.test.js]:
```
'@test'['Type and wait in iframe in an obsolete manner'] = {
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by georgiy-abbasov** [comment]
@helen-dikareva, there was a little mistake with this name, correct name should be `t171129-index.htm`. Do you mean, `t171129.html` is enough for "single-page" tests ?
@AlexanderMoskovkin, in that case these tests will be ungrouped in travis log. They will lose common `describe('Regression tests')` description.

**Comment by AlexanderMoskovkin** [comment]
@georgiy-abbasov  We separate them in the file structure.
Anyway, gh-414 and gh-418 are regression too but they are not grouped in log. There are no difference between them and new tests. If regression test is failed you see `/regression/` in its path in the log, there is no problem with this

**Comment by helen-dikareva** [comment]
I agree with @AlexanderMoskovkin structure and e.g in subfolder 'T171129' page can be just 'index.html'

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
lgtm

**Comment by georgiy-abbasov** [comment]
@AlexanderMoskovkin, FPR

File: test/functional/fixtures/regression/prevent-real-action/testcafe-fixtures/index.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+'@test'['Type and wait in iframe in an obsolete manner'] = {
+    '1.Type in input': function () {
+        var input = $('#input', $('#iframe').contents());
+
+        act.type(input, 'Hello, world!');
+    },
+
+    '2.Wait and perform native click': inIFrame('#iframe', function () {
+        act.wait(5000, function () {
```
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/regression/prevent-real-action/testcafe-fixtures/index.test.js]: We shoud correct this test. I'll explain in private

File: test/functional/fixtures/regression/t171129/test.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+it('T171129 - MVCxImageAndDataNavigationDemos/ImageSlider/VideoGallery demo page', function () {
```
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/regression/t171129/test.js]: Why have you cut the ticket name? It's `T171129 - Test running is hung on the MVCxImageAndDataNavigationDemos/ImageSlider/VideoGallery demo page`

File: test/functional/fixtures/regression/t174562/test.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+it('T174562 - jQuery UI selectmenu demo', function () {
```
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/regression/t174562/test.js]: The same. You should leave the original ticket name, or put a short description of the problem to the test name.
`jQuery UI selectmenu demo` says nothing. You can call it:
`Click on jQuery UI popup item is out of popup bounds (T174562)`

File: test/functional/fixtures/regression/t171129/test.js (hunk 3 above)
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/regression/t171129/test.js]: You can call it something like `test hangs on the MVCxImageSlider demo page (T171129)`

Comment (AlexanderMoskovkin) [inline test/functional/fixtures/regression/t171129/test.js]: Or just leave the origin ticket name ('T171129 - ...')

File: test/functional/site/server.js [hunk 5]
```diff
@@ -48,11 +48,19 @@ Server.prototype._setupRoutes = function () {
     this.app.get('*', function (req, res) {
         var reqPath      = req.params[0] || '';
         var resourcePath = path.join(server.basePath, reqPath);
+        var delay        = typeof req.query.delay === 'number' && parseInt(req.query.delay, 10);
```
Comment (AlexanderMoskovkin) [inline test/functional/site/server.js]: it's not good when a variable can have different types. Now it's `boolean` or `number`.
It should be number or null|number.
If delay is string you can write
```js
    var delay = req.query.delay ? parseInt(req.query.delay) : 0;
```
If it's a number just
```js
var delay = req.query.delay : 0; // as it was in the v.15.1
```

And you can get rid of unnecessary condition `if (delay) ` below

**Comment by georgiy-abbasov** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

/cc @VasilyStrelyaev

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by helen-dikareva** [comment]
lgtm again

**Pull request merged**